stdlib/universe/filter.go

```diff
@@ -345,15 +319,6 @@ func (t *filterTransformation) filter(cr flux.ColReader, record values.Object, i
 	return bitset, nil
 }
 
-func (t *filterTransformation) wait(ctx context.Context, done <-chan struct{}) error {
-	select {
-	case <-done:
-		return nil
-	case <-ctx.Done():
-		return ctx.Err()
-	}
-}
-
```

Comment:
I'm still not clear why waiting for a downstream transformation would cause a deadlock: 

> the dispatcher which, for simple queries with one root, would only
allocate a single goroutine for executing the query. This meant that
since the filter() call never yielded...

Was this for cases where there is no transformation after the call to `filter()`?